brayns/Brayns.cpp

```diff
@@ -115,49 +123,90 @@ struct Brayns::Impl
         if (!_engine)
             throw std::runtime_error("Unsupported engine: " + engineName);
 
+        _engine->getFrameBuffer().clear();
+        _engine->recreate = std::bind(&Impl::createEngine, this);
+        _engine->buildScene = std::bind(&Impl::buildScene, this);
+        _setupCameraManipulator(CameraMode::inspect);
+
         // Default sun light
         DirectionalLightPtr sunLight(
             new DirectionalLight(DEFAULT_SUN_DIRECTION, DEFAULT_SUN_COLOR,
                                  DEFAULT_SUN_INTENSITY));
         _engine->getScene().addLight(sunLight);
+        _engine->getScene().commitLights();
 
-        // Load data and build geometry
         buildScene();
-
-        _engine->recreate = std::bind(&Impl::createEngine, this);
-
-        BRAYNS_INFO << "Now rendering..." << std::endl;
     }
 
     void buildScene()
     {
-        _meshLoader.clear();
+        _engine->setReady(false);
+
+        // if engine does not support scene reloading, recreate it first
+        if (!_engine->getScene().supportsUnloading())
+            createEngine();
+
+        _engine->setLastOperation("");
+        _engine->setLastProgress(0);
+
+        auto loadFunc = [&] {
```

Comment:
make this a _loadScene member function (or free function?), buildScene() is already too long